zjit/src/backend/lir.rs

```diff
@@ -1604,6 +1604,35 @@ impl Assembler
                         Opnd::mem(64, SCRATCH_OPND, 0)
                     };
                     self.incr_counter(counter_opnd, 1.into());
+
+                    use crate::stats::Counter;
+                    let counter = match reason {
+                        SideExitReason::UnknownNewarraySend(_) => Counter::specific_exit_unknown_newarray_send,
+                        SideExitReason::UnknownCallType => Counter::specific_exit_unknown_call_type,
+                        SideExitReason::UnknownOpcode(_) => Counter::specific_exit_unknown_opcode,
+                        SideExitReason::UnhandledInstruction(_) => Counter::specific_exit_unhandled_instruction,
+                        SideExitReason::FixnumAddOverflow => Counter::specific_exit_fixnum_add_overflow,
+                        SideExitReason::FixnumSubOverflow => Counter::specific_exit_fixnum_sub_overflow,
+                        SideExitReason::FixnumMultOverflow => Counter::specific_exit_fixnum_mult_overflow,
+                        SideExitReason::GuardType(_) => Counter::specific_exit_guard_type_failure,
+                        SideExitReason::GuardBitEquals(_) => Counter::specific_exit_guard_bit_equals_failure,
+                        SideExitReason::PatchPoint(_) => Counter::specific_exit_patchpoint,
+                        SideExitReason::CalleeSideExit => Counter::specific_exit_callee_side_exit,
+                        SideExitReason::ObjToStringFallback => Counter::specific_exit_obj_to_string_fallback,
+                        SideExitReason::UnknownSpecialVariable(_) => Counter::specific_exit_unknown_special_variable,
+                        SideExitReason::UnhandledDefinedType(_) => Counter::specific_exit_unhandled_defined_type,
+                        SideExitReason::Interrupt => Counter::specific_exit_interrupt,
+                    };
```

Comment:
How about making this match a function that takes `SideExitReason` as a receiver?